src/inference.jl

```diff
@@ -589,22 +594,43 @@ function inference(;
 
         p = showprogress ? ProgressMeter.Progress(_iterations) : nothing
 
-        for iteration in 1:_iterations
-            inference_invoke_callback(callbacks, :before_iteration, fmodel, iteration)
-            inference_invoke_callback(callbacks, :before_data_update, fmodel, data)
-            for (key, value) in fdata
-                update!(vardict[key], value)
-            end
-            inference_invoke_callback(callbacks, :after_data_update, fmodel, data)
+        record_actors=[]
+        record_fe_actor=[]
+        error_type=[]
+
+        try
+            for iteration in 1:_iterations
+                #sleep allow user interuption
+                if allow_failed
+                    #sleep sometime let the interruption happens inside interation loop
+                    sleep(0.5)
```

Comment:
You don't want to sleep to catch the interruption.
It should go away.

Reply:
I will think about that. If sleep is not allowed, there is a chance that interruptions may occur outside of the iterations, users cannot get the intermediate results. I will think about how to avoid that.